### PR TITLE
Remove hard numpy dep introduced by _inductor/utils.py

### PR DESCRIPTION
**Pull request by amjames**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #90716

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

**Comment by amjames** [comment]
cc @cpuhrsch, should help with blocks on #88078

**Review by cpuhrsch — APPROVED** [review]

File: torch/_inductor/utils.py [hunk 1]
```diff
@@ -117,8 +116,8 @@ def timed(model, example_inputs, times=1):
 
 
 def print_performance(fn, args=(), times=10, repeat=10, baseline=1.0):
-    timings = [timed(fn, args, times) for _ in range(repeat)]
-    took = np.median(timings)
+    timings = torch.tensor([timed(fn, args, times) for _ in range(repeat)])
+    took = torch.median(timings)
```
Comment (vadimkantorov) [inline torch/_inductor/utils.py:120]: just curious, whether it introduces problems with printing below? e.g. can a scalar tensor be formatted directly with `f"{scalar_tensor:.6f}"` or does it need to say `f"{float(scalar_tensor):.6f}"` to convert a scalar tensor to a float?

Comment (amjames) [reply torch/_inductor/utils.py:120]: Seems that the scalar tensor can be formatted just fine.

**Comment by nikitaved** [comment]
Looks like dynamo has more deps:
```
(pytorch-cuda-dev-nik) nik@qgpu3:~/git/Quansight/pytorch/torch/_dynamo$ git grep "import numpy"
allowed_functions.py:import numpy
guards.py:import numpy as np
optimizations/backends.py:import numpy as np
utils.py:import numpy as np
variables/builder.py:import numpy as np
variables/builtin.py:import numpy as np
variables/torch.py:import numpy
```

**Pull request closed without merging**